### ISSUE TITLE
no gameover condition

### ISSUE THREAD

**Issue by dhuang612** [Arcaster42/rhymus#17]
Hey,

So I played the game and took a look at what else could be improved. 
I noticed there is currently no way to lose the game or get a hint to show. 
Maybe creating a counter that keeps track of the number of times the user inputs an incorrect answer and storing it would be a good start for this?

**Comment by Arcaster42** [comment]
That sounds like a great idea! You are correct that the hints are unimplemented now. Let's track the number of incorrect answers, and then have a hint display after one or two incorrect answers (you can decide if it should be one or two).

**Comment by dhuang612** [comment]
Okay, so hints and number of incorrect answers is now setup. 
Next up is wiring up gameover function?

**Comment by Arcaster42** [comment]
Correct. So we have to decide how to do game over. How many chances should the player get? And if you guess right, do you get a chance back?

**Comment by dhuang612** [comment]
>  For example, if a user gets "Game Over", they can still enter the correct answer and move on to the next card. I'll work on that myself, too. Good work!

Hey,

Ya I noticed that too, I wanted to talk about ways we can stop that?
My first thought for that is creating a reset function that pauses the game so that values can't be updated. Then making a restart function?

**Comment by Arcaster42** [comment]
I think we want to avoid making a player refresh. So after game over, we can have a delayed function using `setTimeout` and then after two seconds or so, we can reset the game to its original state from the first card.

**Comment by dhuang612** [comment]
Hey, 
I've been poking around a bit on setting up additional functionality for the gameOver function to do a true reset remove the event listeners and start from card one again.

I originally tried this:
```
const removeListeners = () => {
    guessElement.removeEventListener("keydown", (e) => {
        if (e.key === 'Enter') checkAnswer()
    })
}
```

```
const gameOver = () => {
    if (incorrectGuesses > 4) {
        rhymeElement.textContent = 'Game Over!'
   }
    setTimeout(() => {
        cardBlock.className = 'rhyme_card'
        countElement.className = 'rhyme_count'
        guessElement.className = ''
    }, 2000)
removeListeners()
}
```
Which didn't work.
Through research I learned more about how this was setup as an anonymous function so it'll be more difficult to remove. 
https://www.w3schools.com/jsref/met_element_removeeventlistener.asp
Anonymous functions, like "element.removeEventListener("event", function(){ myScript });" will not work.
